src/sentry/profiles/task.py

```diff
@@ -1189,3 +1200,100 @@ def track_latest_sdk(project: Project, profile: Profile) -> None:
         # unable to acquire the lock means another event is trying to update the version
         # so we can skip the update from this event
         pass
+
+
+@metrics.wraps("process_profile.process_vroomrs_profile")
+def _process_vroomrs_profile(profile: Profile, project: Project) -> bool:
+    if "profiler_id" in profile and _process_vroomrs_chunk_profile(profile=profile):
+        return True
+    _track_failed_outcome(profile, project, "profiling_failed_vroomrs_processing")
+    return False
+
+
+def _process_vroomrs_chunk_profile(profile: Profile) -> bool:
+    with sentry_sdk.start_span(op="task.profiling.process_vroomrs_chunk_profile"):
+        try:
+            # todo (improvement): check the feasibility of passing the profile
+            # dict directly to the PyO3 module to avoid json serialization/deserialization
+            with sentry_sdk.start_span(op="json.dumps"):
+                json_profile = json.dumps(profile)
+            with sentry_sdk.start_span(op="json.unmarshal"):
+                chunk = vroomrs.profile_chunk_from_json_str(json_profile, profile["platform"])
+            chunk.normalize()
+            with sentry_sdk.start_span(op="gcs.write", name="compress and write"):
+                storage = get_profiles_storage()
+                compressed_chunk = chunk.compress()
+                storage.save(chunk.storage_path(), io.BytesIO(compressed_chunk))
+            with sentry_sdk.start_span(op="processing", name="send chunk to kafka"):
+                payload = build_chunk_kafka_message(chunk)
+                topic = ArroyoTopic(get_topic_definition(Topic.PROFILE_CHUNKS)["real_topic_name"])
+                profile_chunks_producer.produce(topic, payload)
+            with sentry_sdk.start_span(op="processing", name="extract functions metrics"):
+                functions = chunk.extract_functions_metrics(
+                    min_depth=1, filter_system_frames=True, max_unique_functions=100
+                )
+                payload = build_chunk_functions_kafka_message(chunk, functions)
+                topic = ArroyoTopic(
+                    get_topic_definition(Topic.PROFILES_CALL_TREE)["real_topic_name"]
+                )
+                profile_functions_producer.produce(topic, payload)
+            return True
+        except Exception as e:
```

Comment:
This has no retries built-in as far as I can tell. Is this something we should look into?

Reply:
I'm quite surprised this was not added to the `SingletoneProducer` options in Arroyo.

I've followed up with the team maintaining Arroyo [here](https://sentry.slack.com/archives/C04E4723EJ1/p1746451747689219), and I proposed to add the retries, Backoff.ms, maxBackoff.ms and delivery.timeout.ms to the list of allowed options. 

Let's see what they say.

Reply:
So, long story short: it has built-in retries for the simple fact that it's a wrapper around rdkafka, which already has default value for retries, minBackoff, maxBackoff, delivery.timeout etc. etc. see here for the default value: https://docs.confluent.io/platform/current/clients/librdkafka/html/md_CONFIGURATION.html

With that said I still think we should add them as configurable option in `SUPPORTED_KAFKA_CONFIGURATION` so that if we or anyone else want to specialize those options for their own producer they can, but I would do this in a separate PR has it has nothing to do with this PR and, as for now, I think we're safe to rely on the defaults as for all the other running producers in Sentry.

Reply:
Im more curious about retries on GCS

Reply:
Yes, the GCS filestore implementation does support retries.